librecad/src/lib/generators/lc_makercamsvg.cpp

```diff
@@ -543,6 +550,193 @@ void LC_MakerCamSVG::writeEllipse(RS_Ellipse* ellipse) {
+}
+
+void LC_MakerCamSVG::writeCubicBeziers(const std::vector<RS_Vector> &control_points, bool is_closed) {
+
+    std::vector<RS_Vector> bezier_points = calcCubicBezierPoints(control_points, is_closed);
+
+    std::string path = svgPathMoveTo(convertToSvg(bezier_points[0]));
+
+    int bezier_points_size = bezier_points.size();
+
+    int bezier_count = ((bezier_points_size - 1) / 3);
+
+    for (int i = 0; i < bezier_count; i++) {
+        path += svgPathCurveTo(convertToSvg(bezier_points[3 * (i + 1)]), convertToSvg(bezier_points[3 * (i + 1) - 2]), convertToSvg(bezier_points[3 * (i + 1) - 1]));
+    }
+
+    xmlWriter->addElement("path", NAMESPACE_URI_SVG);
+
+    xmlWriter->addAttribute("d", path);
+
+    xmlWriter->closeElement();
+}
+
+void LC_MakerCamSVG::writeQuadraticBeziers(const std::vector<RS_Vector> &control_points, bool is_closed) {
+
+    std::vector<RS_Vector> bezier_points = calcQuadraticBezierPoints(control_points, is_closed);
+
+    std::string path = svgPathMoveTo(convertToSvg(bezier_points[0]));
+
+    int bezier_points_size = bezier_points.size();
+
+    int bezier_count = ((bezier_points_size - 1) / 2);
+
+    for (int i = 0; i < bezier_count; i++) {
+        path += svgPathQuadraticCurveTo(convertToSvg(bezier_points[2 * (i + 1)]), convertToSvg(bezier_points[2 * (i + 1) - 1]));
+    }
+
+    xmlWriter->addElement("path", NAMESPACE_URI_SVG);
+
+    xmlWriter->addAttribute("d", path);
+
+    xmlWriter->closeElement();
+}
+
+std::vector<RS_Vector> LC_MakerCamSVG::calcCubicBezierPoints(const std::vector<RS_Vector> &control_points, bool is_closed) {
+
+    std::vector<RS_Vector> bezier_points;
+
+    int control_points_size = control_points.size();
+
+    int bezier_points_size;
+
+    if (is_closed) {
+
+        for (int i = 0; i < (control_points_size - 1); i++) {
+            bezier_points.push_back(control_points[i]);
+
+            bezier_points.push_back((control_points[i] * 2.0 + control_points[i + 1]) / 3.0);
+            bezier_points.push_back((control_points[i] + control_points[i + 1] * 2.0) / 3.0);
+        }
+
+        bezier_points.push_back(control_points[control_points_size - 1]);
+        bezier_points.push_back((control_points[control_points_size - 1] * 2.0 + control_points[0]) / 3.0);
+        bezier_points.push_back((control_points[control_points_size - 1] + control_points[0] * 2.0) / 3.0);
+        bezier_points.push_back(control_points[0]);
+
+        // Auxiliary points for easier calculation
+        bezier_points.insert(bezier_points.begin(), ((control_points[control_points_size - 1] + control_points[0] * 2.0) / 3.0));
+        bezier_points.push_back((control_points[0] * 2.0 + control_points[1]) / 3.0);
+
+        bezier_points_size = bezier_points.size();
+
+        for (int i = 1; i < (bezier_points_size - 1); i += 3) {
+            bezier_points[i] = ((bezier_points[i - 1] + bezier_points[i + 1]) / 2.0);
+        }
+
```

Comment:
splinepoints class holds a quadratic spline passing given list of points(control points).

I feel the new cubic bezier won't pass precisely those initial control points any more.